tests/cases/fourslash/codeFixMapIndexSignature.ts

```diff
@@ -0,0 +1,12 @@
+/// <reference path='fourslash.ts' />
+
+// @target:ES2015
+// @strict:true
+
+// target and lib do not actually work in four slash tests, so we define Map manually
+//// declare class Map<K, V> { get(key: string): V; }
+////[|let map = new Map<string, string>();
+////map['af123']|];
```

Comment:
Since you seem like the type to go for bonus points, may I suggest that you try this example?
```ts
/*a*/map/*b*/[/*c*/'a'/*d*/]/*e*/ //f
```
Frankly, I wouldn't worry too much if you don't like the results - most of our code fixes don't handle this gracefully (cc @jessetrinity).

Reply:
Yeah, most of those comments get removed, except for the end of line comment